src/Mvc/Mvc.Core/test/ModelBinding/FormValueProviderTest.cs

```diff
@@ -9,6 +9,31 @@ namespace Microsoft.AspNetCore.Mvc.ModelBinding;
 
 public class FormValueProviderTest : EnumerableValueProviderTest
 {
+    [Fact]
+    public void GetValue_ReturnsInvariantCulture_IfInvariantEntryExists()
+    {
+        // Arrange
+        var culture = new CultureInfo("fr-FR");
+        var invariantCultureKey = "prefix.name";
+        var currentCultureKey = "some";
+        var values = new Dictionary<string, StringValues>(BackingStore)
+        {
+            { FormValueProvider.CultureInvariantFieldName, new(invariantCultureKey) },
+        };
+        var valueProvider = GetEnumerableValueProvider(BindingSource.Query, values, culture);
+
+        // Act
+        var invariantCultureResult = valueProvider.GetValue(invariantCultureKey);
+        var currentCultureResult = valueProvider.GetValue(currentCultureKey);
+
+        // Assert
+        Assert.Equal(CultureInfo.InvariantCulture, invariantCultureResult.Culture);
+        Assert.Equal(BackingStore[invariantCultureKey], invariantCultureResult.Values);
+
```

Comment:
What do you think about split into 2 tests?

- `GetValue_ReturnsInvariantCulture_IfInvariantEntryExists`
- `GetValue_ReturnsCurrentCulture_IfIfInvariantEntryNotFound `-> you can find a better name 😁

Reply:
We could do that, but the reason I had them included in the same test was so there are multiple form entries present, where some use culture-invariant formatting and others don't. Splitting it up into two tests would lose this detail.

Reply:
Got it.